apisix/plugins/hmac-auth.lua

```diff
@@ -224,6 +224,13 @@ local function generate_signature(ctx, secret_key, params)
 
         for _, key in pairs(keys) do
             local param = args[key]
+            -- when args without `=<value>`, value is treated as true.
+            -- In order to be compatible with args lacking `=<value>`,
+            -- we need to replace true with an empty string.
+            if type(param) == "boolean" then
+                param = ""
```

Comment:
Have you investigated implementations in other languages like `node.js`, `golang`, `python`?

Reply:
I tested it, and the following is my test result：

node.js: curl "http://localhost:8080/args?name=jack&age"

```
node.js, args name：jack
node.js, args age：
```

python: curl "http://localhost:8080/args?name=jack&age"

```
ParseResult(scheme='http', netloc='127.0.0.1:8080', path='/args', params='', query='name=jack&age', fragment='')
{'age': [''], 'name': ['jack']}
```

golang: curl  "http://localhost:8080/args?name=jack&age"

```
map[age:[] name:[jack]]
args value:  jack 
```